src/bot.ts

```diff
@@ -82,6 +83,7 @@ export class DiscordBot {
     /* Handles messages queued up to be sent to matrix from discord. */
     private discordMessageQueue: { [channelId: string]: Promise<void> };
     private channelLock: Lock<string>;
+    private typingTimers: Record<string, NodeJS.Timeout> = {}; // DiscordUser+channel -> Timeout
```

Comment:
why not just set synapse timeout to 30s?

Reply:
TIL that's a thing.